src/Dma.h

```diff
@@ -107,6 +113,8 @@ namespace RTX {
     
     // business logic
     virtual int allocateDemandToJunctions(time_t time);
+    std::shared_ptr<DemandAllocationDelegate> getAllocationDelegate();
```

Comment:
convention in this codebase is to omit "get" from getters - simply `allocationDelegate` and `setAllocationDelegate`

Reply:
yeaaa i  wasn't sure because of `getProjectionString()` in Model.cpp

Reply:
![image](https://github.com/user-attachments/assets/6a25a786-1068-43f1-bded-0a57de500945)
@cameron-devine you in trouble! 😆

Reply:
Oops.... 😆

Reply:
i'll fix that in this pr